compute/scheduler/scheduler.go

```diff
@@ -63,11 +63,17 @@ func (s *Scheduler) CheckNodes() error {
 
 		if node.State == NodeState_GONE {
 			for _, tid := range node.TaskIds {
-				s.Event.WriteEvent(ctx, events.NewState(tid, tes.State_SYSTEM_ERROR))
-				s.Event.WriteEvent(ctx, events.NewSystemLog(tid, 0, 0, "info",
+				err = s.Event.WriteEvent(ctx, events.NewState(tid, tes.State_SYSTEM_ERROR))
+				if err != nil {
+					return err
```

Comment:
Similar to my previous comments you might check if these writers are wrapped with the ErrorLogger. 

It might be a better approach to update the ErrorLogger so that it doesn’t return errors so you don’t need to make these sort of changes at all (you'd have to create another interface though and update usages accordingly).

One other option I'll throw out - add a specific exclusion for Writer.Write calls for the errcheck linter (see: https://golangci-lint.run/usage/false-positives/#specific-linter-excludes).

Reply:
Taking a look at updating the `events.ErrLogger` (or possibly just adding an exclusion to the linter). Tending toward adding the exclusion, reverting the `err` changes, and creating an issue to track adding a new interface.

Reply:
Sorry I left so many unhandled errors! I wrote this back when I was still learning go, so hindsight is 20/20 I guess 😄

Reply:
All good! I'm pretty new to the world of Go so have lots to learn in regards to it's error management and best practices. It's been nice learning from this code base!